cmd/eksctl-anywhere/cmd/createcluster.go

```diff
@@ -143,10 +162,30 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command, _ []string) er
 		return err
 	}
 
+	if _, err = time.ParseDuration(cc.timeoutOptions.cpWaitTimeout); err != nil {
+		logger.V(0).Info(fmt.Sprintf(timeoutWarningTemplate,
+			cpWaitTimeoutFlag, cc.timeoutOptions.cpWaitTimeout, clustermanager.DefaultControlPlaneWaitStr))
+		cc.timeoutOptions.cpWaitTimeout = clustermanager.DefaultControlPlaneWaitStr
+	}
+
+	if _, err = time.ParseDuration(cc.timeoutOptions.externalEtcdWaitTimeout); err != nil {
+		logger.V(0).Info(fmt.Sprintf(timeoutWarningTemplate,
+			externalEtcdWaitTimeoutFlag, cc.timeoutOptions.externalEtcdWaitTimeout, clustermanager.DefaultEtcdWaitStr))
+		cc.timeoutOptions.externalEtcdWaitTimeout = clustermanager.DefaultEtcdWaitStr
+	}
+
+	perMachineWaitTimeout, err := time.ParseDuration(cc.timeoutOptions.perMachineWaitTimeout)
+	if err != nil {
+		logger.V(0).Info(fmt.Sprintf(timeoutWarningTemplate,
+			perMachineWaitTimeoutFlag, cc.timeoutOptions.perMachineWaitTimeout, clustermanager.DefaultPerMachineWaitStr))
+		perMachineWaitTimeout = clustermanager.DefaultMaxWaitPerMachine
+	}
```

Comment:
Can we move this parsing logic somewhere else and share it between commands? It could even be a method in `timeoutOptions`

Reply:
Yes will do